### PR TITLE
Properly throw errors on users management

### PR DESCRIPTION
**Pull request by skjnldsv**
The list of error codes is a mess btw, should we fix them as well?
What is the proper way of doing this? One error code per error type? Bad email 102, bad quota, 103.. etc?

### CODE REVIEW

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 1]
```diff
@@ -458,7 +459,7 @@ public function editUser(string $userId, string $key, string $value): DataRespon
 
 		$targetUser = $this->userManager->get($userId);
 		if ($targetUser === null) {
-			throw new OCSException('', \OCP\API::RESPOND_UNAUTHORISED);
+			throw new OCSException('Unknown user', 101);
```
Comment (skjnldsv) [inline apps/provisioning_api/lib/Controller/UsersController.php]: to be consistent with others methods on this page :man_shrugging:

**Comment by kesselb** [comment]
> The list of error codes is a mess btw

At least its [documented](https://docs.nextcloud.com/server/16/admin_manual/configuration_user/instruction_set_for_users.html#edit-data-of-a-single-user) ;)

**Comment by skjnldsv** [comment]
> At least its [documented](https://docs.nextcloud.com/server/16/admin_manual/configuration_user/instruction_set_for_users.html#edit-data-of-a-single-user) ;)

Lol, wrongly! It says `101 - user not found`, but it's actually not what is in the code! :scream:

**Review by blizzz — APPROVED** [review]

File: apps/provisioning_api/lib/Controller/UsersController.php (hunk 1 above)
Comment (nickvergessen) [reply apps/provisioning_api/lib/Controller/UsersController.php]: But now it reveals to anyone if a user exists, while before it required it to be your own user or you having subadmin/admin permissions

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 2]
```diff
@@ -638,7 +639,7 @@ public function deleteUser(string $userId): DataResponse {
 		$targetUser = $this->userManager->get($userId);
 
 		if ($targetUser === null || $targetUser->getUID() === $currentLoggedInUser->getUID()) {
-			throw new OCSException('', 101);
+			throw new OCSException('Unknown user', 101);
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/UsersController.php]: unknown is not correct for the second part of the if condition

**Comment by MorrisJobke** [comment]
Nothing for 20 -> move to 21

**Pull request closed without merging**